bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/HomeConnectBridgeHandler.java

```diff
@@ -0,0 +1,295 @@
+    private static final String CLIENT_ID = "clientId";
+
+    private final OAuthFactory oAuthFactory;
+    private final HomeConnectServlet homeConnectServlet;
+    private final Logger logger;
+
+    private @Nullable ScheduledFuture<?> reinitializationFuture;
+    private @Nullable List<ApiRequest> apiRequestHistory;
+    private @Nullable List<Event> eventHistory;
+
+    private @NonNullByDefault({}) OAuthClientService oAuthClientService;
+    private @NonNullByDefault({}) String oAuthServiceHandleId;
+    private @NonNullByDefault({}) HomeConnectApiClient apiClient;
+    private @NonNullByDefault({}) HomeConnectEventSourceClient eventSourceClient;
+
+    public HomeConnectBridgeHandler(Bridge bridge, OAuthFactory oAuthFactory, HomeConnectServlet homeConnectServlet) {
+        super(bridge);
+
+        this.oAuthFactory = oAuthFactory;
+        this.homeConnectServlet = homeConnectServlet;
+        logger = LoggerFactory.getLogger(HomeConnectBridgeHandler.class);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // not used for bridge
+    }
+
+    @Override
+    public void initialize() {
+        Version version = FrameworkUtil.getBundle(this.getClass()).getVersion();
+        String openHabVersion = OpenHAB.getVersion();
+        logger.debug("Initialize bridge {} (Bundle: {}, openHAB: {})", getThing().getLabel(), version.toString(),
+                openHabVersion);
+        // let the bridge configuration servlet know about this handler
+        homeConnectServlet.addBridgeHandler(this);
+
+        // create oAuth service
+        ApiBridgeConfiguration config = getConfiguration();
+        String tokenUrl = (config.isSimulator() ? API_SIMULATOR_BASE_URL : API_BASE_URL) + OAUTH_TOKEN_PATH;
+        String authorizeUrl = (config.isSimulator() ? API_SIMULATOR_BASE_URL : API_BASE_URL) + OAUTH_AUTHORIZE_PATH;
+        String oAuthServiceHandleId = thing.getUID().getAsString() + (config.isSimulator() ? "simulator" : "");
+
+        oAuthClientService = oAuthFactory.createOAuthClientService(oAuthServiceHandleId, tokenUrl, authorizeUrl,
+                config.getClientId(), config.getClientSecret(), OAUTH_SCOPE, true);
+        this.oAuthServiceHandleId = oAuthServiceHandleId;
+        logger.debug(
+                "Initialize oAuth client service. tokenUrl={}, authorizeUrl={}, oAuthServiceHandleId={}, scope={}, oAuthClientService={}",
+                tokenUrl, authorizeUrl, oAuthServiceHandleId, OAUTH_SCOPE, oAuthClientService);
+
+        // create api client
+        apiClient = new HomeConnectApiClient(oAuthClientService, config.isSimulator(), apiRequestHistory);
+        eventSourceClient = new HomeConnectEventSourceClient(oAuthClientService, config.isSimulator(), scheduler,
+                eventHistory);
+
+        try {
+            @Nullable
+            AccessTokenResponse accessTokenResponse = oAuthClientService.getAccessTokenResponse();
```

Comment:
All the code than can take time should be run in a separate thread to be sure to have a fast run of initialize()